src/tools/perf/api/libperf.h

```diff
@@ -97,6 +97,19 @@ enum {
 };
 
 
+enum {
```

Comment:
I reflected a bit on this and discussed with Mikhail.
IMO it still makes sense to keep all these IDs in the same enum, to make sure the values are unique. Because all these AM IDs are used to register daemon event handlers, so they must be distinct and enum enforces that.

At the same time I agree, for clarity reason it's nice to be able to distinguish what is what.
So I changed this enum into:
```
typedef enum {
    /* Host to daemon communication */
    UCP_PERF_DAEMON_AM_ID_HOST_INIT     = 0,
    UCP_PERF_DAEMON_AM_ID_HOST_REQ      = 1,
    UCP_PERF_DAEMON_AM_ID_HOST_SEND_ACK = 2,
    UCP_PERF_DAEMON_AM_ID_HOST_RECV_ACK = 3,
    UCP_PERF_DAEMON_AM_ID_HOST_FIN      = 4,

    /* Peer to peer daemon communication */
    UCP_PERF_DAEMON_AM_ID_PEER_INIT     = 5,
    UCP_PERF_DAEMON_AM_ID_PEER_OP       = 6
} ucp_perf_daemon_am_id_t;
```
Does it make sense?

Reply:
Yeah, it definitely make sense for me. WDYT about the following proposal?

```c
typedef enum {
    UCP_PERF_DAEMON_AM_ID_HOST_INIT,
    UCP_PERF_DAEMON_AM_ID_HOST_REQ,
    UCP_PERF_DAEMON_AM_ID_HOST_SEND_ACK,
    UCP_PERF_DAEMON_AM_ID_HOST_RECV_ACK,
    UCP_PERF_DAEMON_AM_ID_HOST_FIN,
} ucp_perf_daemon_am_id_host_t;

typedef enum {
    UCP_PERF_DAEMON_AM_ID_PEER_INIT = UCP_PERF_DAEMON_AM_ID_HOST_FIN + 1,
    UCP_PERF_DAEMON_AM_ID_PEER_OP
} ucp_perf_daemon_am_id_peer_t;
```

Reply:
Personally I don't see much benefit, it will rather complicate the updated handler logic (which is not in PR yet):
```
// I'm using this enum type for registering handlers on supported messages:
static void
ucp_perf_daemon_set_am_recv_handler(ucp_perf_daemon_context_t *ctx,
                                    ucp_perf_daemon_am_id_t am_id, << Common ID
                                    ucp_am_recv_callback_t cb)

Which is then called for both host-peer and peer-peer communication:

+    ucp_perf_daemon_set_am_recv_handler(ctx, UCP_PERF_DAEMON_AM_ID_HOST_INIT,
+                                        ucp_perf_daemon_host_init_handler);
+    ucp_perf_daemon_set_am_recv_handler(ctx, UCP_PERF_DAEMON_AM_ID_PEER_INIT,
+                                        ucp_perf_daemon_peer_init_handler);

```
So I would keep it in the same enum, because it's used uniformly for the same stuff.
IMO splitting it further does not bring much benefits (unless I miss them?), but makes code longer and more involved. Anyway, if at some point we decide to split this enum - we can easily do that in minutes